### PR TITLE
CHANGELOG: update for 0.10

### PR DESCRIPTION
**Pull request by Profpatsch**
Towards https://github.com/tweag/rules_haskell/issues/1040

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,19 +6,90 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/).
...
+
+* `haskell_cabal_library`/`haskell_cabal_binary`: Both use `cabal` to
+  build Haskell package dependencies.
+
+* `stack_snapshot`: Uses stack’s dependency resolving algorithm to
+  generate `haskell_cabal_library` targets automatically.
+  Requires `stack` > 2.1, which it will download if cannot be found in
+  `PATH`.
+
+* It is now possible to statically link Haskell libraries in CC
```
Comment (aherrmann) [inline CHANGELOG.md]: Just to clarify for myself, this is referring to 0ff1690525a676725401081bc646189378071c40?

File: CHANGELOG.md [hunk 2]
```diff
@@ -168,16 +283,6 @@ of `rules_haskell`.
   sed -i 's/^haskell_import/haskell_toolchain_library/' **/BUILD{,.bazel}
   sed -i 's/"haskell_import"/"haskell_toolchain_library"/' **/BUILD{,.bazel}
   ```
-* The `haskell_toolchain` macro now no longer adds a `toolchain`
```
[the remark this answers is not in the document]
Comment (Profpatsch) [reply CHANGELOG.md]: Yeah, that tripped me up, too. It was added after the previous release, in an oversight.

File: CHANGELOG.md (hunk 1 above)
Comment (Profpatsch) [reply CHANGELOG.md]: Yes, did I interpret this commit message incorrectly?

Comment (aherrmann) [reply CHANGELOG.md]: Yes, looks good to me.

File: CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (aherrmann) [reply CHANGELOG.md]: I see, thanks for explaining.

**Pull request merged**